### ISSUE TITLE
DatabaseSchemaIntrospection.get_constraints() can return incorrect values for Django 1.11+

### ISSUE THREAD

**Issue by ajmatsick** [bernardopires/django-tenant-schemas#523]
With the addition of class-based indexes in Django 1.11, some changes were made to the way index and constraint data is retrieved and returned in `django/db/backends/postgresql/introspection.py`.

See:
* [Issue #25809](https://code.djangoproject.com/ticket/25809)
* [Issue #27135](https://code.djangoproject.com/ticket/27135)

Of note:
* Each constraint `dict` in `DatabaseIntrospection.get_constraints()` now includes an `options` key
* The `type` value is now `django.db.models.indexes.Index.suffix` for instead of `btree` for B-tree indexes

As far as I can tell the omission of the `options` key doesn't currently have an effect, though it may in the future when using class-based indexes. The incorrect `type` value, however, can cause errors when changing the `db_index` and `unique` values for model fields.

For example, consider a model field that has an index and is then modified to add a unique constraint:

```python
class Migration(migrations.Migration):
    operations = [
        migrations.CreateModel(
            name='Example',
            fields=[
                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
                ('name', models.CharField(max_length=100, db_index=True)),
            ],
            options={
            },
            bases=(models.Model,),
        ),

        migrations.AlterField(
            model_name='Example',
            name='name',
            field=models.CharField(
                max_length=100,
                db_index=True,
                unique=True,
            ),
        )
    ]
```

This migration will fail with an error similar to this:

> django.db.utils.ProgrammingError: relation "app_example_name_f63d8867_like" already exists

This is because the existing index names are not collected correctly (see [django/db/backends/base/schema.py](https://github.com/django/django/blob/1decd0197d241b27d54bb12eca04b7e89a9ccba6/django/db/backends/base/schema.py#L578)), meaning the existing index is not removed before being added again later ([here](https://github.com/django/django/blob/1decd0197d241b27d54bb12eca04b7e89a9ccba6/django/db/backends/base/schema.py#L716)).

To fix this, `tenant_schemas/postgresql_backend/introspection.py` should be updated to reflect the changes in `django/db/backends/postgresql/introspection.py`.

**Comment by ajmatsick** [comment]
Unfortunately I checked the open issues before submitting this, but not the open pull requests. I just saw that #519 addresses the same thing. I can close this if that's going to be merged instead.

**Comment by AGASS007** [comment]
whatever gets this fix going!

**Comment by ajmatsick** [comment]
Fixed by #519
